core/test/stage_mockups.h

```diff
@@ -128,5 +129,13 @@ struct TaskTestBase : public testing::Test
 	}
 };
 
+#define EXPECT_COSTS(value, matcher)                                           \
+	{                                                                           \
+		std ::vector<double> costs;                                              \
+		std::transform(value.begin(), value.end(), std::back_inserter(costs),    \
+		               [](const SolutionBaseConstPtr& s) { return s->cost(); }); \
+		EXPECT_THAT(costs, matcher);                                             \
+	}
+
```

Comment:
Nice simplification, though I'm no big friend of preprocessor macros... Do functions hide the line from gtests output?

Reply:
Yes. That's the reason gtest and gmock use macros everywhere. I share your dislike of macros :wink: